astropy/io/fits/hdu/hdulist.py

```diff
@@ -510,6 +510,10 @@ def fileinfo(self, index):
 
         return output
 
+    def copy(self):
+        # Return an HDUList when calling HDUList.copy() (issue #7211)
+        return self[:]
```

Comment:
This returns a view, not a deep copy. It is unclear whether `copy()` should return a shallow or deep copy. I'll let the maintainers decide.

Reply:
It seems to me like the expected behavior would be a deep copy, meaning that the HDU objects in each of the lists should be equal but different. But @saimn should weigh in as well.

Reply:
I think it should be a shallow copy, because `list.copy` and `copy.copy(hdulist)` do shallow copies, and also because it can be useful to copy just the container for instance to add a new HDU.

But there is also another discussion in #6395 about this and `copy.deepcopy` which do not work when a HDUList is read from a file (because by default it inherits `list` which tries to serialize the object).